UI/src/app/courses/components/courses-page/courses-page.component.ts

```diff
@@ -23,26 +26,36 @@ export class CoursesPageComponent implements OnInit {
   filter = new BehaviorSubject<string>('');
   //filter = new Observable<string>();
 
+  courses$: Observable<any>;
+
   private filterSubscrition: Subscription;
+  private deleteSubscrition: Subscription;
 
   constructor (
-    public coursesStoreService: CoursesStoreService,
+    // public coursesStoreService: CoursesStoreService,
     private coursesBackendService: CourseService,
-    private router: Router
+    private router: Router,
+    private store: Store<CoursesState>
   ) {}
 
   ngOnInit(): void {
     this.filterSubscrition = this.filter
     .pipe(
-      debounceTime(200)
+      debounceTime(200),
+      tap(value => {
+        if(!value.length || value.length >= 3) {
```

Comment:
To be more declarative we could use `filter` operator instead of `if`

same for almost all stream parts with if statement